src/credentials.rs

```diff
@@ -0,0 +1,105 @@
+use crate::errors::SdError;
+use nix::dir;
+use nix::fcntl::OFlag;
+use nix::sys::stat::Mode;
+use std::env;
+use std::fs::File;
+use std::path::PathBuf;
+
+/// Credential loader for units.
+///
+/// Credentials are read by systemd on unit startup and exported by their ID.
+///
+/// **Note**: only the user associated with the unit and the superuser may access credentials.
+///
+/// More documentation: <https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Credentials>
+#[derive(Debug)]
+pub struct CredentialsLoader {
+    path: PathBuf,
+    _dirfd: dir::Dir,
```

Comment:
I presume this is a safe nix type with a proper Drop implementation which closes the FD again?

Sorry if that's a stupid question; I'm on mobile and can't easily check right now 😊

Reply:
Yes, it automatically performs a `closedir()` on Drop: https://docs.rs/nix/0.25.0/src/nix/dir.rs.html#88-95